deepcell/applications/application.py

```diff
@@ -0,0 +1,164 @@
+
+        # Test that pre and post processing functions are callable
+        if (self.preprocessing_fn is not None) and (callable(self.preprocessing_fn) is False):
+            raise ValueError('Preprocessing_fn must be a callable function.')
+        if (self.postprocessing_fn is not None) and (callable(self.postprocessing_fn) is False):
+            raise ValueError('Postprocessing_fn must be a callable function.')
+
+    def predict(self, x):
+        raise NotImplementedError
+
+    def _predict_segmentation(self,
+                              image,
+                              batch_size=4,
+                              image_mpp=None,
+                              preprocess_kwargs={},
+                              postprocess_kwargs={},
+                              debug=False):
+        """Generates a labeled image of the input running prediction with
+        appropriate pre and post processing functions.
+
+        Input images are required to have 4 dimensions [batch, x, y, channel]. Additional
+        empty dimensions can be added using `np.expand_dims`
+
+        Args:
+            image (np.array): Input image with shape [batch, x, y, channel]
+            batch_size (int, optional): Number of images to predict on per batch. Defaults to 4.
+            image_mpp (float, optional): Microns per pixel for the input image. Defaults to None.
+            preprocess_kwargs (dict, optional): Kwargs to pass to preprocessing function.
+                Defaults to {}.
+            postprocess_kwargs (dict, optional): Kwargs to pass to postprocessing function.
+                Defaults to {}.
+            debug (bool, optional): If True, returns intermediate outputs of data processing.
+                Defaults to False.
+
+        Raises:
+            ValueError: Input data must have 4 dimensions, [batch, x, y, channel
+
+        Returns:
+            np.array: Labeled image, if debug is False.
+                If debug is True, returns (image, tiles, output_tiles, output_images, label_image)
+        """
+
+        # Check input size of image
+        if len(image.shape) != self.required_rank:
+            raise ValueError('Input data must have {} dimensions'
+                             'Input data only has {} dimensions'.format(str(self.required_rank), str(len(image.shape))))
+
+        # Resize image if necessary
+        if (image_mpp is not None) & (image_mpp != self.model_mpp):
+            original_shape = image.shape
+            scale_factor = image_mpp / self.model_mpp
+            new_shape = (int(image.shape[1] / scale_factor),
+                         int(image.shape[2] / scale_factor))
+            image = resize(image, new_shape, data_format='channels_last')
+        else:
+            original_shape = None
+
+        # Preprocess image
+        if self.preprocessing_fn is not None:
+            image = self.preprocessing_fn(image, **preprocess_kwargs)
+            print('preprocessing')
```

Comment:
I would prefer not to have these print statements. If necessary, use a logger.

Reply:
My bad, left over from debugging.